store/src/state.rs

```diff
@@ -254,6 +254,20 @@ impl State {
             let _ = mem::replace(&mut inner.account_tree, account_tree);
         }
 
+        if let Err(err) = handle.await {
+            error!(
+                is_cancelled = err.is_cancelled(),
+                is_panic = err.is_panic(),
+                "apply_block task joined with an error"
+            );
+
+            // This should never happen. It is unclear if the state of the node would be valid
+            // because the apply_block task may have failed when commiting the transaction, so the
+            // in-memory and the DB state would be out-of-sync.
```

Comment:
AFAIK the join would error in two circumstances:

- The task was cancelled somehow
- The task panicked 

A SQL error is not a panic, so it would not appear here. I'm going to add another log line for that.

Regardless, I'm still pretty sure we should not have a SQL error in there :)